include/seqan/arg_parse/arg_parse_type_support.h

```diff
@@ -126,6 +165,15 @@ inline bool isIntegerArgument(ArgParseArgument const & me);
 inline bool isInt64Argument(ArgParseArgument const & me);
 inline bool isDoubleArgument(ArgParseArgument const & me);
 inline bool isStringArgument(ArgParseArgument const & me);
+inline bool isBooleanArgument(ArgParseArgument const & me);
+
+inline bool _convertArgumentValue(bool & dst, ArgParseArgument const & opt, std::string const & src)
+{
+    if (!isBooleanArgument(opt))
+        return false;
+
+    return _castBool(dst, src);
```

Comment:
There if only [one generic _tryCast() function](https://github.com/seqan/seqan/blob/master/include/seqan/arg_parse/arg_parse_type_support.h#L51) that handles every cast so far. Since the boolean functionality does not actually cast the string directly to a bool, I thought to keep the difference in this behaviour by a different name instead of overloading the function _tryCast() for boolean type.
Maybe the name _tryCastFromList or _tryCastFromEquivalenceClass or some other would fit better?